env/iasConfig.go

```diff
@@ -46,17 +52,103 @@ func GetIASConfig() (*Identity, error) {
 		}
 		return &vcapServices.Identity[0].Credentials, nil
 	case kubernetes:
-		return nil, fmt.Errorf("unable to parse ias config: kubernetes env detected but not yet supported")
+		var secretPath = os.Getenv(iasConfigPathKey)
+		if secretPath == "" {
+			secretPath = iasConfigPathDefault
+		}
+		identities, err := readServiceBindings(secretPath)
+		if err != nil || len(identities) == 0 {
+			return nil, fmt.Errorf("cannot find '%s' service binding from secret path '%s'", iasServiceName, secretPath)
+		} else if len(identities) > 1 {
+			return nil, fmt.Errorf("found more than one '%s' service instance from secret path '%s'. This is currently not supported", iasServiceName, secretPath)
+		}
+		return &identities[0], nil
 	default:
-		return nil, fmt.Errorf("unable to parse ias config: unknown environment detected")
+		return nil, fmt.Errorf("unable to parse '%s' service config: unknown environment detected", iasServiceName)
+	}
+}
+
+func readServiceBindings(secretPath string) ([]Identity, error) {
+	instancesBound, err := ioutil.ReadDir(secretPath)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read service directory '%s' for identity service: %w", secretPath, err)
+	}
+	identities := []Identity{}
+	for _, instanceBound := range instancesBound {
+		if !instanceBound.IsDir() {
+			continue
+		}
+		serviceInstancePath := path.Join(secretPath, instanceBound.Name())
+		instanceSecretFiles, err := ioutil.ReadDir(serviceInstancePath)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read service instance directory '%s' for '%s' service instance '%s': %w", serviceInstancePath, iasServiceName, instanceBound.Name(), err)
+		}
+		instanceSecretsJSON, err := readCredentialsFileToJSON(serviceInstancePath, instanceSecretFiles)
+		if instanceSecretsJSON == nil || err != nil {
+			instanceSecretsJSON, err = readSecretFilesToJSON(serviceInstancePath, instanceSecretFiles)
+			if err != nil {
+				return nil, err
+			}
+		}
+		identity := Identity{}
+		if err := json.Unmarshal(instanceSecretsJSON, &identity); err != nil {
+			return nil, fmt.Errorf("cannot unmarshal json content in directory '%s' for '%s' service instance: %w", serviceInstancePath, iasServiceName, err)
+		}
+		identities = append(identities, identity)
+	}
+	return identities, nil
+}
+
+func readCredentialsFileToJSON(serviceInstancePath string, instanceSecretFiles []fs.FileInfo) ([]byte, error) {
+	for _, instanceSecretFile := range instanceSecretFiles {
+		if !instanceSecretFile.IsDir() && instanceSecretFile.Name() == iasSecretKeyDefault {
+			serviceInstanceCredentialsPath := path.Join(serviceInstancePath, instanceSecretFile.Name())
+			credentials, err := ioutil.ReadFile(serviceInstanceCredentialsPath)
+			if err != nil {
+				return nil, fmt.Errorf("cannot read content from '%s': %w", serviceInstanceCredentialsPath, err)
+			}
+			if json.Valid(credentials) {
```

Comment:
Nit: I think this can be omitted, since we unmarshal a few lines later and we'll fail there if the content is invalid.

Reply:
You are right. But this json.Valid check i require for the case that someone has not specified "secretKey" (service operator) but has a secretFile named "credentials" with no json input, then it will continue to map all secret files....
I have a test case for the same.